Directory.Build.targets

```diff
@@ -10,17 +10,19 @@
     <!-- Workaround while there is no 5.0 SDK available, copy known apphost/framework reference info from 3.0 -->
     <KnownAppHostPack 
       Include="@(KnownAppHostPack->WithMetadataValue('TargetFramework', 'netcoreapp3.0'))"
```

Comment:
I couldn't figure out a good way to generalize this to "Include KnownAppHostPack items where TargetFramework != DefaultNetCoreTargetFramework" - as far as I can tell, WithMetadataValue is inclusive only: https://docs.microsoft.com/en-us/visualstudio/msbuild/item-functions?view=vs-2019

Reply:
Combine `WithMetadataValue` and `Count` item functions

Reply:
I'm not sure that works either - I can include items based on `WithMetadataValue`, but I could only use `Count` in a condition. I could include all items in `KnownFrameworkReference` here, then condition on: `KnownFrameworkReference` has no Items with MetadataValue `TargetFramework` = `DefaultNetCoreTargetFramework`, AND `TargetFramework` starts with `Netcoreapp`, but then if we used a 5.0 SDK in vNext (6.0?), we'd be adding 6.0 entries for both the 3.0 & 5.0 netcoreapp references in the 5.0 SDK, which seems wrong.

Reply:
```
Condition=" @(KnownAppHostPack->WithMetadataValue('TargetFramework', '$(DefaultNetCoreTargetFramework)->Count() == 0 "
```
&hellip;
```
AND %(KnownAppHost.TargetFramework) == 'netcoreapp3.0'
```
and use `$(DefaultNetCoreTargetFramework)` in the added item.

Reply:
In that case, I think this is already in the desired state - We add an entry that duplicates the `netcoreapp3.0` entry, but with `TargetFramework = $(DefaultNetCoreTargetFramework)`, only when the ItemGroup doesn't already have an item with `TargetFramework = $(DefaultNetCoreTargetFramework)`.